consensus/src/metrics_safety_rules.rs

```diff
@@ -125,3 +148,121 @@ impl TSafetyRules for MetricsSafetyRules {
         })
     }
 }
+
+pub struct MockSafetyRules {
+    // number of initialize() calls
+    init_calls: i32,
+
+    // max initialize() calls to complete perform_initialize()
+    max_init_calls: i32,
+}
+
+impl MockSafetyRules {
+    #[allow(dead_code)]
```

Comment:
If you move this into the test module, you should be able to remove this line.

Reply:
Moved it inside mod tests.